inc/hclib-task.h

```diff
@@ -32,32 +32,20 @@ typedef struct _hclib_task_t {
      * and locality flexible asyncAny
      */
     int is_async_any_type;
```

Comment:
I'm trying to get this merged in now that I've realized each of Nick's diffs includes all of the preceding diffs. Can we delete this is_async_any_type now? It really does not belong in hclib.